sql/core/src/test/scala/org/apache/spark/sql/SubquerySuite.scala

```diff
@@ -1765,4 +1765,35 @@ class SubquerySuite extends QueryTest with SharedSparkSession with AdaptiveSpark
       }
     }
   }
+
+  test("SPARK-34946: correlated scalar subquery in grouping expressions only") {
```

Comment:
Could you move these tests into `AnalysisErrorSuite`?

Reply:
That test suite needs to build logical plans using DSL and the test would be hard to read with subqueries.

Reply:
It seems it has some tests using a SQL string, e.g., 
https://github.com/apache/spark/blob/7c8dc5e0b54731c5790bb94274eb9308774772ff/sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/AnalysisErrorSuite.scala#L184-L187

Reply:
ah I see, then +1!

Reply:
Agree. And it would also be easier to check for all subquery-related errors if we have the tests in one test suite.